packages/dds/tree/src/feature-libraries/editable-tree/editableTreeSequence.ts

```diff
@@ -0,0 +1,382 @@
+    }
+
+    public getType(key?: string, nameOnly?: boolean): TreeSchemaIdentifier | TreeSchema | undefined {
+        let typeName: TreeSchemaIdentifier | undefined = this.cursor.type;
+        const primaryKey = this.primaryKey;
+        if (key !== undefined) {
+            if (primaryKey !== undefined) {
+                const result = this.cursor.down(primaryKey, 0);
+                if (result === TreeNavigationResult.Ok) {
+                    typeName = this.cursor.seek(Number(key)) === TreeNavigationResult.Ok ? this.cursor.type : undefined;
+                    this.cursor.up();
+                }
+            } else {
+                typeName = this.cursor.seek(Number(key)) === TreeNavigationResult.Ok ? this.cursor.type : undefined;
+            }
+        }
+        if (nameOnly) {
+            return typeName;
+        }
+        if (typeName) {
+            return lookupTreeSchema(this.context.forest.schema, typeName);
+        }
+        return undefined;
+    }
+
+    /**
+     * Sets value of the node if index exists.
+     */
+    public setValue(index: number, value: Value): boolean {
+        const primaryKey = this.primaryKey;
+        assert(primaryKey !== undefined, "Not supported");
+        assert(this.cursor.down(primaryKey, index) === TreeNavigationResult.Ok, "Cannot navigate to a node to set value");
+        const target = this.context.createTarget(this.cursor);
+        this.cursor.up();
+        const type = target.getType() as TreeSchema;
+        assert(isPrimitive(type), `"Set value" is not supported for non-primitive fields`);
+        const path = target.getPath();
+        assert(path !== undefined, "Can't locate a path to set a value");
+        return this.context.setNodeValue(path, value);
+    }
+
+    /**
+     * Appends a node to a sequence.
+     */
+    public appendNode(cursor: ITreeCursor): boolean {
+        const primaryKey = this.primaryKey;
+        const length = this.length;
+        assert(primaryKey !== undefined, "Not supported");
+        const path = this.getPath();
+        if (this.context.insertNode({
+            parent: path,
+            parentField: primaryKey,
+            parentIndex: length,
+        }, cursor)) {
+            const result = this.cursor.down(primaryKey, length);
+            this.push(this.context.createTarget(this.cursor));
+            this.cursor.up();
+            return result === TreeNavigationResult.Ok;
+        }
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    // TODO: (?) return deleted nodes as a detached array sequence
+    splice(start: number, deleteCount?: number, ...items: ProxyTarget[]): ProxyTarget[] {
+        throw new Error("Not implemented");
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    // TODO: (?) create a copy of a sequence slice as a detached array sequence
+    slice(start?: number, end?: number): ProxyTarget[] {
+        throw new Error("Not implemented");
+    }
+
+    /**
+     * {@inheritdoc}
```

Comment:
Since docs are inherited from parent types implicitly, this tag is only really worth including when you need to inherit docs from something specific  that is not actually inferable here. We have been omitting is in these cases.